### PR TITLE
Re-added ACTWebSocket support

### PR DESCRIPTION
**Pull request by Rawrington**
I have re-added ACTWebSocket support while maintaining OverlayPlugin compatibility, this build works with both ACTWS and OverlayPlugin from github pages, and does not have any underlying issues, I also fixed an issue wherein limit breaks and other strange things would cause a combatant to get cut off at the end of the list. Additionally I have updated package.json to work under the latest NPM with a bunch of updated dependancies.

### CODE REVIEW

**Comment by Rawrington** [comment]
NOTE: This has been tested with both plugins on my end through github pages
(https://rawrington.github.io/horizoverlay/)

**Comment by Rawrington** [comment]
(Currently doing a bug fix as I noticed an issue)

**Comment by Rawrington** [comment]
testing it now, should be fixed though.

**Comment by bsides** [comment]
Awesome, can't wait to test it too! Will give it a try tonight to push this ASAP. 
Thank you so much for your efforts!!!

File: package.json [hunk 1]
```diff
@@ -9,23 +9,23 @@
...
     "type": "git",
-    "url": "https://github.com/bsides/horizoverlay.git"
+    "url": "https://github.com/Rawrington/horizoverlay.git"
   },
   "bugs": {
-    "url": "https://github.com/bsides/horizoverlay/issues"
+    "url": "https://github.com/Rawrington/horizoverlay/issues"
   },
-  "homepage": "https://bsides.github.io/horizoverlay",
+  "homepage": "https://rawrington.github.io/horizoverlay",
```
Comment (bsides) [inline package.json]: The reference from the original repo should still be the same. Could you make this change?

File: package.json [hunk 2]
```diff
@@ -37,5 +37,17 @@
...
+    "production": [
+      ">0.2%",
+      "not dead",
+      "not op_mini all"
+    ],
+    "development": [
+      ">0.2%",
+      "not dead",
+      "not op_mini all"
+    ]
```
Comment (bsides) [inline package.json]: I believe we should check which version the Overlay plugin uses so we can deploy with that target in mind. I'll let you know in my tests tonight which version is it. It'd be better to other developers to not be confused with API changes.

File: src/Combatants.js [hunk 3]
```diff
@@ -12,7 +12,12 @@ class Combatants extends Component {
   render() {
     const maxRows = this.props.config.maxCombatants
     const dataArray = Object.keys(this.props.data)
-    const battler = dataArray.slice(0, maxRows)
+	const regexp = /[a-zA-Z]+\s+[a-zA-Z]+/g;
+    const battler = dataArray.filter(player => (
```
Comment (bsides) [inline src/Combatants.js]: Just one little thing here: do you have prettier installed in your IDE? If not, I'll set a thing later to make this better indented once we push our files (ie. a post-commit hook).

File: package.json [hunk 4]
```diff
@@ -9,12 +9,12 @@
...
     "type": "git",
-    "url": "https://github.com/Rawrington/horizoverlay.git"
+    "url": "https://github.com/bsides/horizoverlay.git"
   },
   "bugs": {
-    "url": "https://github.com/Rawrington/horizoverlay/issues"
+    "url": "https://github.com/bsides/horizoverlay/issues"
   },
-  "homepage": "https://rawrington.github.io/horizoverlay",
+  "homepage": "https://bsides.github.io/horizoverlay",
```
Comment (bsides) [inline package.json]: Oh well forget about my last comment about this 😅

File: src/Combatants.js (hunk 3 above)
Comment (bsides) [reply src/Combatants.js]: ...and you've done it here. Reviews should begin with the last ones, heh!

File: package.json (hunk 1 above)
Comment (Rawrington) [reply package.json]: yeah I'll sort it

File: package.json (hunk 2 above)
Comment (Rawrington) [reply package.json]: yeah true, this specific combination of requirements DOES work with overlayplugin which is why i chose it

File: package.json (hunk 1 above)
Comment (Rawrington) [reply package.json]: oh it was sorted

File: src/Combatants.js (hunk 3 above)
Comment (Rawrington) [reply src/Combatants.js]: i actually fixed spacing in a later thing :P

**Comment by Rawrington** [comment]
I've changed how the slicing works in my personal build and cut out jobless/outputless party members to help with party filter settings in ACT, I'm thinking of implementing multiple rows so people can show the full 24 people if they want on 24 mans (i.e. a rows setting in config)

**Comment by Rawrington** [comment]
I can push this to my be merged branch if you want me to :P

**Comment by Rawrington** [comment]
my latest commit should sort out all the issues, i added in my 2 filters, you can change the show jobless combatants option in config to put the behaviour back to how it originally behaved :P

**Comment by bsides** [comment]
> I can push this to my be merged branch if you want me to :P

Looks nice, did you make it as an option? These days have been crazy to test it though 😭

**Comment by bsides** [comment]
Although I cannot test it, I'll just accept this. If anything goes wrong we can fix or just throw back the older version.

**Pull request merged**